expression/aggregation/base_func.go

```diff
@@ -269,6 +283,16 @@ func (a *baseFuncDesc) typeInfer4MaxMin(ctx sessionctx.Context) {
 		(a.Name != ast.AggFuncFirstRow && a.Name != ast.AggFuncMax && a.Name != ast.AggFuncMin) {
 		a.RetTp = &types.FieldType{Tp: mysql.TypeString, Flen: mysql.MaxFieldCharLength}
 	}
+	// valid the precision and scale of decimal type
+	if a.RetTp.Tp == mysql.TypeNewDecimal {
+		a.RetTp = a.RetTp.Clone()
```

Comment:
If the input type is decimal(23,-1), then it will be changed to decimal(23,30)?

Reply:
Does this take affect for all the functions? For example, currently, if we try to add implicit cast from double to decimal, the return type of the cast function will be decimal(23,-1), so after this pr, it will be updated to decimal(23,30)?

Reply:
https://github.com/pingcap/tidb/issues/30347 let it as another issue.